lib/ddtrace/contrib/sinatra/request_span_middleware.rb

```diff
@@ -0,0 +1,92 @@
+module Datadog
+  module Contrib
+    module Sinatra
+      # Middleware used for automatically tagging configured headers and handle request span
+      class RequestSpanMiddleware
+        SINATRA_REQUEST_SPAN = 'datadog.sinatra_request_span'.freeze
+        SINATRA_REQUEST_TRACE_NAME = 'sinatra.request'.freeze
+
+        def initialize(app)
+          @app = app
+        end
+
+        def call(env)
+          span = request_span!(env)
+          _, headers, = @app.call(env)
+        ensure
```

Comment:
I see this `ensure` here, and `set_error` below in `tracer.rb`. Are we appropriately setting failed requests with errors? And where should this be done?

Reply:
Hmm, good question. Tracer uses Sinatra facilities to process error, which I think is appropriate in this context.